tests/events.rs

```diff
@@ -0,0 +1,35 @@
+use std::time::Duration;
+
+use mio::{Events, Poll, Token, Waker};
+
+const WAKE_TOKEN: Token = Token(10);
+
+#[test]
+fn events_all() {
+    let mut events = Events::with_capacity(16);
+    assert_eq!(events.capacity(), 16);
+    assert!(events.is_empty());
+
+    let mut poll = Poll::new().expect("unable to create new Poll instance");
+    let token = Token(10);
+    let waker = Waker::new(poll.registry(), token).expect("unable to create waker");
+
+    waker.wake().expect("unable to wake");
+    poll.poll(&mut events, Some(Duration::from_millis(100)))
+        .unwrap();
+
+    assert!(!events.is_empty());
+
+    for event in events.iter() {
+        assert_eq!(event.token(), token);
+        assert!(event.is_readable());
+    }
+
+    assert!(!events.is_empty());
```

Comment:
This block tests the same thing as the for loop above.

Reply:
removed